examples/hotrod/pkg/tracing/mux.go

```diff
@@ -40,37 +40,28 @@ func NewServeMux(copyBaggage bool, tracer opentracing.Tracer, logger log.Factory
 // TracedServeMux is a wrapper around http.ServeMux that instruments handlers for tracing.
 type TracedServeMux struct {
 	mux         *http.ServeMux
-	copyBaggage bool
-	tracer      opentracing.Tracer
+	tracer      trace.TracerProvider
 	logger      log.Factory
+	req         http.Request
 }
 
 // Handle implements http.ServeMux#Handle, which is used to register new handler.
 func (tm *TracedServeMux) Handle(pattern string, handler http.Handler) {
 	tm.logger.Bg().Debug("registering traced handler", zap.String("endpoint", pattern))
 
-	middleware := nethttp.Middleware(
-		tm.tracer,
-		handler,
-		nethttp.OperationNameFunc(func(r *http.Request) string {
-			return "HTTP " + r.Method + " " + pattern
-		}),
-		// Jaeger SDK was able to accept `jaeger-baggage` header even for requests without am active trace.
-		// OTEL Bridge does not support that, so we use Baggage propagator to manually extract the baggage
-		// into Context (in otelBaggageExtractor handler below), and once the Bridge creates a Span,
-		// we use this SpanObserver to copy OTEL baggage from Context into the Span.
-		nethttp.MWSpanObserver(func(span opentracing.Span, r *http.Request) {
-			if !tm.copyBaggage {
-				return
-			}
-			bag := baggage.FromContext(r.Context())
-			for _, m := range bag.Members() {
-				if b := span.BaggageItem(m.Key()); b == "" {
-					span.SetBaggageItem(m.Key(), m.Value())
-				}
-			}
-		}),
-	)
+	// otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+
+	ctx := tm.req.Context()
```

Comment:
`Handle` function is called at initialization time, not at request handing time. You can put this logic into an HTTP handler, similar to what `otelBaggageExtractor` or `otelhttp.NewHandler` functions are doing. You just need to make sure the ordering is correct, because `middleware` from above was running _after_ the span was created.

It's also not clear to me if you even need this at all. The original reason I had to add `middleware` function was because the baggage was extracted from request via OTEL Propagator and placed into `Context`, but the rest of the code was using OpenTracing instrumentation and was expecting to get the baggage from the `opentracing.Span`, so `middleware` was doing a copy from Context to Span. Since we now are fully migrating to OTEL instrumentation, I think just `otelBaggageExtractor` will be sufficient (as long as we change the code that reads the baggage to read it from Context, not from Span).

We may also not even need `otelBaggageExtractor` eventually, since it was working around the limitation of the Bridge where it was not parsing baggage if the request had no tracing context (as happens between UI and `frontend`). However, with OTEL instrumentation baggage and span creation are decoupled, so I assume baggage will still be parsed correctly by the propagators into Context.

Reply:
Thanks Yuri. It was an great explanation. I am able to extract context from you now. 😄 
> Handle function is called at initialization time, not at request handling time. 

That's why we needed to call this `func (tm *TracedServeMux) Handle(...)` while starting the server.
> It's also not clear to me if you even need this at all.

That means we don't need to create this for baggage propagator https://github.com/jaegertracing/jaeger/blob/e427e196ec409ce577ae2acb1fc4e640f580f68f/examples/hotrod/pkg/tracing/mux.go#L54-L60 since we have OTEL instrumentation does this job and maybe not eventually `otelBaggageExtractor`.